### PR TITLE
[WIP] Don't use regular CSRF token

### PR DESCRIPTION
**Pull request by LukasReschke**
When handling external content using the regular CSRF token is a risky action as one misconfiguration (i.e. the HTML parser would not work properly) would leak the CSRF token to linked content.

Thus this includes a custom CSRF implementation.

Todo:
- [x] Write remaining tests
- [ ] Fix remaining TODOs / FIXMEs

### CODE REVIEW

File: lib/controller/messagescontroller.php [hunk 1]
```diff
@@ -124,6 +130,7 @@ public function index($from=0, $to=20, $filter=null) {
 	/**
 	 * @NoAdminRequired
 	 * @NoCSRFRequired
+	 * @UseSession
```
Comment (LukasReschke) [inline lib/controller/messagescontroller.php]: FIXME: Hacky workaround. Needs to get assigned earlier to allow closing session here.

Comment (LukasReschke) [inline lib/controller/messagescontroller.php]: Move to index controller of the app

File: lib/controller/proxycontroller.php [hunk 2]
```diff
@@ -93,12 +102,9 @@ public function redirect() {
...
-	public function proxy() {
-		// close the session to allow parallel downloads
-		$this->session->close();
-
-		$resourceURL = $this->request->getParam('src');
-		$content =  \OC::$server->getHelper()->getUrlContent($resourceURL);
-		return new ProxyDownloadResponse($content, $resourceURL, 'application/octet-stream');
+	public function proxy($src) {
+		$src = urldecode($src);
+		$content = \OC::$server->getHelper()->getUrlContent($src);
```
Comment (LukasReschke) [inline lib/controller/proxycontroller.php]: While at it we can also inject this one…

File: lib/message.php [hunk 3]
```diff
@@ -49,10 +52,13 @@ function __construct($conn, $mailBox, $messageId, $fetch=null,
 		$this->messageId = $messageId;
 		$this->loadHtmlMessage = $loadHtmlMessage;
 
-		// TODO: inject ???
-//		$cacheDir = \OC::$server->getUserFolder() . '/mail/html-cache';
-//		$this->htmlService = new Html($cacheDir);
-		$this->htmlService = new Html();
+		// FIXME: Inject
```
Comment (LukasReschke) [inline lib/message.php]: TODO

**Comment by jancborchardt** [comment]
@LukasReschke I’m moving this to 0.3 and let you party on Mallorca :D

**Comment by ChristophWurst** [comment]
@LukasReschke what's the status on this? :-)

**Comment by ChristophWurst** [comment]
@LukasReschke ping

**Comment by ChristophWurst** [comment]
No response, so I'm moving this to the backlog for now.

**Comment by ChristophWurst** [comment]
No activity for more than a year -> close

**Pull request closed without merging**